fluss-server/src/test/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManagerTest.java

```diff
@@ -0,0 +1,423 @@
+        for (int i = 0; i < 3; i++) {
+            RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", i);
+            manager.registerCollector(collector);
+            testCollectors.add(collector);
+            collectors.add(collector);
+        }
+
+        assertThat(manager.isRunning()).isTrue();
+        assertThat(manager.getCollectorCount()).isEqualTo(3);
+
+        // Unregister all collectors
+        for (RocksDBMetricsCollector collector : testCollectors) {
+            manager.unregisterCollector(collector);
+        }
+
+        assertThat(manager.getCollectorCount()).isEqualTo(0);
+        // Manager should still be running (collection thread continues)
+        assertThat(manager.isRunning()).isTrue();
+    }
+
+    @Test
+    void testResourceCleanupOnShutdown() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        // Register a collector to start the manager
+        RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", 0);
+        manager.registerCollector(collector);
+        collectors.add(collector);
+
+        assertThat(manager.isRunning()).isTrue();
+
+        // Get the executor service via reflection to verify it's cleaned up
+        Field executorField =
+                RocksDBMetricsManager.class.getDeclaredField("metricsCollectionExecutor");
+        executorField.setAccessible(true);
+        ScheduledExecutorService executor = (ScheduledExecutorService) executorField.get(manager);
+
+        assertThat(executor.isShutdown()).isFalse();
+
+        manager.shutdown();
+
+        assertThat(manager.isShutdown()).isTrue();
+        assertThat(executor.isShutdown()).isTrue();
+    }
+
+    private RocksDBMetricsCollector createMockCollector(String database, String table, int bucket) {
+        return createMockCollectorWithPartition(database, table, bucket, null);
+    }
+
+    private RocksDBMetricsCollector createMockCollectorWithPartition(
+            String database, String table, int bucket, String partition) {
+        Statistics mockStatistics = mock(Statistics.class);
+
+        TablePath tablePath = TablePath.of(database, table);
+        TableBucket tableBucket = new TableBucket(0, bucket);
+
+        return spy(
+                new RocksDBMetricsCollector(
+                        rocksDBExtension.getRocksDb(),
+                        mockStatistics,
+                        TestingMetricGroups.BUCKET_METRICS,
+                        tableBucket,
+                        tablePath,
+                        partition,
+                        resourceContainer));
+    }
+
+    private void resetSingleton() {
+        try {
+            Field instanceField = RocksDBMetricsManager.class.getDeclaredField("instance");
+            instanceField.setAccessible(true);
+            instanceField.set(null, null);
+        } catch (Exception e) {
```

Comment:
Usually, reflection is highly not recommended, unless it is absolutely necessary. Here we can declare the `instance` variable as `protected` to make it can be accessed in the `RocksDBMetricsManagerTest` but not exposed publicly.